src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1487,6 +1499,22 @@ private static bool IsTransparentIdentifierFieldInAnonymousType(FieldSymbol fiel
             }
         }
 
+        private static bool TryParseGeneratedName(string name, out GeneratedNameKind kind, out string part)
```

Comment:
>bool [](start = 23, length = 4)

Since no call site is using the bool this could just by void ParseGeneratedName, or ```(kind, part) ParseGeneratedName(name)```